assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js

```diff
@@ -140,7 +140,10 @@ export const PaymentMethodDataProvider = ( { children } ) => {
 				id: 'wc-express-payment-error',
 			} );
 		} else {
-			removeNotice( 'wc-express-payment-error' );
+			removeNotice(
+				'wc-express-payment-error',
+				'wc/express-payment-area'
```

Comment:
Curious about how this `removeNotice` API works - does the context parameter clear out all notices in that context (i.e. clear out all express payment notices), or is it more like a namespace, so `wc-express-payment-error` is only removed if it's in `wc/express-payment-area` context?

If it's the former, can we simplify the API so it clears a whole notice context - i.e. `clearAllNotices( 'wc/express-payment-area' )`?

Related, are there multiple contexts in play on the checkout, and do we want to clear them all on every submit? E.g. `clearAllNotices()` ?

Reply:
You can see the `removeNotice()` signature here:
https://github.com/WordPress/gutenberg/blob/master/packages/notices/src/store/actions.js#L147

Basically, you pass the id of the notice to remove and (optionally) its context.

In `payment-method-data-context.js `, we are adding notices to different contexts, so when removing them we need to pass the context as well.

> Related, are there multiple contexts in play on the checkout, and do we want to clear them all on every submit? E.g. `clearAllNotices()` ?

We have a system in place to remove all notices from the current context:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/hooks/use-store-notices.js#L40-L47

Used here, for example:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/base/context/cart-checkout/checkout-state/index.js#L182

There isn't an easy way to remove all notices from another context than the current one. And in fact, I'm not sure we would want to do that even if it was possible. I imagine, for example, a `Subscribe to our newsletter` block in the sidebar that used the notices API as well. It might show an error notice when the wrong email is introduced, but we don't want that notice to disappear when submitting _Checkout_ for example, because they have no relation between.